Migrate schedule.json to TOML

psvenk opened the pull request:
Closes #253. This helps readability and maintainability considerably. Some changes have been made to the frontend to work with native JavaScript `Date` objects instead of numbers.

psvenk committed:
Migrate schedule.json to TOML

This helps readability and maintainability considerably. Some changes
have been made to the frontend to work with native JavaScript Date
objects instead of numbers.

notrodes commented:
Can you add an error throw as a default case in the switch statement on line 178 of clock.js?

tektaxi reviewed (approved):
Works fine for me, but I haven't looked through the code much.

psvenk commented:
@notrodes Are you referring to the `update_lunch` function? That is currently dormant code which will be addressed in #95 / #288 and is out of scope here. Are there any issues with the code that I wrote/touched in this pull request, or is this ready to merge?

notrodes commented on public/js/clock.js:
```diff
@@ -29,6 +29,18 @@ let date_override = undefined;
+    for (const [, schedule] of Object.entries(schedules)) {
+        for (entry of schedule) {
+            // Convert each entry to a JavaScript Date object
+            let [, hours, minutes, seconds, ms]
```
Can you avoid this regex repeat?

psvenk replied:
Fixed in 2571fc5157.

psvenk committed:
Deduplicate regex handling

Avoid repeating the regex for extracting components from a time, and
make this part of the code more concise in general.
